cmd/juju/application/deploy.go

```diff
@@ -369,6 +369,42 @@ Constraints can be specified by specifying the '--constraints' option. If the
 application is later scaled out with ` + "`juju add-unit`" + `, provisioned machines
 will use the same constraints (unless changed by ` + "`juju set-constraints`" + `).
 
+Application configuration values can be specified using '--config' option. This
+option accepts either a path to a yaml-formatted file or a key=value pair. 
+Configurtion file provided should be in format
+<charm name>:
+	<option name>: <option value>
+	...
+For example, to deploying 'mediawiki' with the configuration file 'mycfg.yaml'
+that contains:
+
+mediawiki:
+	name: my media wiki
+	admins: me:pwdOne
+	debug: true
+
+use
+
+  juju deploy mediawiki --config mycfg.yaml
+
+To specify key=value pair to set an application option value, use:
+ 
+  juju deploy mediawiki --config name='my media wiki'
+   
+When specifying more than one option value, use:
+
+  juju deploy mediawiki --config name='my media wiki' --config debug=true
+
+Care must be taken when specifying more than one configuration via 
+'--config' option - if duplicate option names are found,
+the last one will be used. The same will apply if both a file path and 
+key=pair(s) are specified.
+For example, when calling
+
+  juju deploy mediawiki --config name='my media wiki' --config mycfg.yaml
+  
+the farthest, i.e. the file, will be used to configure deployed application.
```

Comment:
Not quite right.
name='my media wiki' is still used.

It's just that if mycfg.yaml also contains a value for 'name', that will override the earlier value.
As will
juju deploy foo --config name=fred --config name=mary

Reply:
perhaps also s/farthest/last

Reply:
the example 'mycfg.yaml' contains a 'name' option. So I think it makes sense s is....